agenta-web/src/components/SaveTestsetModal/SaveTestsetModal.tsx

```diff
@@ -0,0 +1,73 @@
+import {Evaluation, EvaluationScenario} from "@/lib/Types"
+import {EvaluationFlow} from "@/lib/enums"
+import {createNewTestset} from "@/lib/services/api"
+import {Form, Input, Modal, message} from "antd"
+import React, {Dispatch, SetStateAction} from "react"
+
+type EvaluationRow = EvaluationScenario & {
+    evaluationFlow: EvaluationFlow
+} & {[variantId: string]: string}
+
+interface SaveTestsetModalProps {
+    appId: string
+    isTestsetModalOpen: boolean
+    setIsTestsetModalOpen: Dispatch<SetStateAction<boolean>>
+    evaluation: Evaluation
+    rows: EvaluationRow[]
+}
+
+const SaveTestsetModal: React.FC<SaveTestsetModalProps> = ({
+    setIsTestsetModalOpen,
+    appId,
+    isTestsetModalOpen,
+    evaluation,
+    rows,
+}) => {
+    const [form] = Form.useForm()
+
+    const handleSave = async () => {
```

Comment:
Please pass this function to Form's `onFinish` callback and read the form values from the funciton's args.